docs/blog/_posts/2019-08-30-18th-dotty-milestone-release.md

```diff
@@ -0,0 +1,318 @@
+
+An astute reader has probably noticed the mentions of things like `CommandLineParser` in the body of the generated method, which hints to certain features. That's right, we support a basic ability for command-line args parsing:
+
+```scala
+@main def sayHello(name: String, age: Int): Unit =
+  println(s"Hello $name, you are $age years old")
+```
+
+If you run the above with command line arguments "Jack 25", the output will be "Hello Jack, you are 25 years old". And here is how you can define a custom parser for your own class:
+
+```scala
+case class Address(city: String, street: String)
+
+given as scala.util.FromString[Address] {
+  /** Can throw java.lang.IllegalArgumentException */
+  def fromString(s: String): T =
+    s.split(",").toList match {
+      case city :: street :: Nil => Address(city, street)
+      case _ => throw new IllegalArgumentException(s"Please specify address in the format 'city, street'")
+    }
+}
+
+@main def sayHello(addr: Address): Unit =
+  println(s"You are living at $addr")
+```
+
+The motivation for the `@main` functions is to make Scala scripting friendly. So far we do not plan to support something more complex than the above – we believe if a user needs a complex command line parsing capability, they can always fall back to the conventional `def main(args: Array[String])` syntax plus a dedicated library like [scopt](https://github.com/scopt/scopt). The changes described above, however, are already enough to make script development much less tedious than before.
+
+To learn more, see the [documentation](http://dotty.epfl.ch/docs/reference/changed-features/main-functions.html).
+
+## Allow infix operators at the start of the line
+A small change yet relevant to many. Now, you can write the following code:
+
+```scala
+def isABorC(x: Char) = x == 'A'
+  || x == 'B'
+  || x == 'C'
+```
+
+Prior to this change, it was only possible to express infix operators at the beginning of the line as follows:
+
+```scala
+def isABorC(x: Char) = (x == 'A'
+  || x == 'B'
+  || x == 'C')
+```
+
+## Drop do-while syntax
+Remember that obscure `do-while` feature of Scala 2 where you could write:
+
+```scala
+scala> var x = 0
+x: Int = 0
+
+scala> val iterator = Iterator.from(10, -1)
+iterator: Iterator[Int] = <iterator>
+
+scala> do {
+     |   x = iterator.next()
+     |   if (x >= 0) println(x)
+     | } while (x >= 0)
+10
+9
+8
+7
+6
+5
+4
+3
+2
+1
+0
+```
+
+Well, it is no more! That is the only place where the `do` token is used in Scala, the feature itself is rarely employed, and it would be nice to reclaim the `do` token for other uses (described in details in the section on the new syntax for control expressions).
+
+We did make `while` more powerful so that the language does not lose its expressiveness. Now, you can write the following to achieve the same functionality:
```

Comment:
You can already write that in Scala 2. It is not new at all.

Reply:
So we didn't add anything here, only removed?

Reply:
That's right.